service/sharddistributor/service.go

```diff
@@ -37,29 +45,76 @@ import (
 
 // Service represents the shard distributor service
 type Service struct {
-	resource.Resource
+	logger        log.Logger
+	metricsClient metrics.Client
+	dispatcher    *yarpc.Dispatcher
 
-	status           int32
 	handler          handler.Handler
-	stopC            chan struct{}
 	config           *config.Config
 	numHistoryShards int
 	peerProvider     membership.PeerProvider
 
-	matchingRing *membership.Ring
-	historyRing  *membership.Ring
+	matchingRing membership.SingleProvider
+	historyRing  membership.SingleProvider
+
+	// DEPRECATED: does not affect fx lifecycle.
+	stopC    chan struct{}
```

Comment:
Can we just remove this? I think it's not needed anymore

Reply:
It is required for legacy mode (where Start() is blocking until stopped)